src/common/SurgeLogger.cpp

```diff
@@ -0,0 +1,57 @@
+#include "SurgeLogger.h"
+#include "spdlog/sinks/basic_file_sink.h"
+#include "spdlog/sinks/stdout_sinks.h"
+
+#include "spdlog/logger.h"
+#include "spdlog/async.h"
+#include <memory>
+
+
+typedef std::shared_ptr< spdlog::logger > log_t;
+
+log_t internal_surge_log_init()
+{
+  std::vector<spdlog::sink_ptr> sinks;
+  
+  /*
+  ** Where to put the logfile is actually OS dependent. 
+  ** The approach we take here is to add a pair of sinks
+  ** to the logger, the first to stderr
+  */
+  auto stderr_logger = std::make_shared<spdlog::sinks::stderr_sink_mt>();
+  sinks.push_back( stderr_logger );
+  
+  /*
+  ** And if someone has implemented this for their OS, the second to a standard
+  ** logfile location
+  */
+
+#if MAC
+  char lfb[ PATH_MAX ];
+  sprintf( lfb, "%s/Library/Logs/Surge.log", getenv( "HOME" ) );
+  auto filesink = make_shared<spdlog::sinks::basic_file_sink_mt> (lfb);
+  sinks.push_back( filesink );
+#endif
+
+#if WINDOWS
+  // FIXME: Where's the default log location on macos?
+#endif
+
+#if __linux__
+  // FIXME: Where's the default log location on linux?
+#endif
+
+
+  /* auto logger = std::make_shared<spdlog::async_logger>("surgelogger", sinks.begin(), sinks.end(),
+                                                       spdlog::thread_pool(),
+                                                       spdlog::async_overflow_policy::block); */
+
```

Comment:
I don't like cruft code to be added to repository. These will just give only WTF moments some time from now.